pkg/sql/sem/tree/datum.go

```diff
@@ -650,7 +650,7 @@ func NewDInt(d DInt) *DInt {
 // ParseDInt parses and returns the *DInt Datum value represented by the provided
 // string, or an error if parsing is unsuccessful.
 func ParseDInt(s string) (*DInt, error) {
-	i, err := strconv.ParseInt(s, 0, 64)
+	i, err := strconv.ParseInt(s, 10, 64)
```

Comment:
so looks like this mostly works, `'9223372036854775808'::int` in strconv here will return the appropriate error. but please add tests.